satpy/readers/agrl_l1.py

```diff
@@ -0,0 +1,182 @@
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Advanced Geostationary Radiation Imager for the Level_1 HDF format
+
+The files read by this reader are described in the official Real Time Data Service:
+
+    http://fy4.nsmc.org.cn/data/en/data/realtime.html
+
+"""
+
+import logging
+import numpy as np
+from datetime import datetime
+from pyresample import geometry
+from satpy.readers.hdf5_utils import HDF5FileHandler
+
+logger = logging.getLogger(__name__)
+
+# info of 500 m, 1 km, 2 km and 4 km data
+_resolution_list = [500, 1000, 2000, 4000]
+_COFF_list = [10991.5, 5495.5, 2747.5, 1373.5]
+_CFAC_list = [81865099.0, 40932549.0, 20466274.0, 10233137.0]
+_LOFF_list = [10991.5, 5495.5, 2747.5, 1373.5]
+_LFAC_list = [81865099.0, 40932549.0, 20466274.0, 10233137.0]
+
+
+class HDF_AGRL_L1(HDF5FileHandler):
+
+    def __init__(self, filename, filename_info, filetype_info):
+        super(HDF_AGRL_L1, self).__init__(filename, filename_info, filetype_info)
+
+    def get_dataset(self, dataset_id, ds_info):
+        """Load a dataset."""
+        logger.debug('Reading in get_dataset %s.', dataset_id.name)
+        file_key = ds_info.get('file_key', dataset_id.name)
+        data = self.get(file_key)
+
+        # convert bytes to string
+        data.attrs['long_name'] = data.attrs['long_name'].decode("utf-8")
+        data.attrs['band_names'] = data.attrs['band_names'].decode("utf-8")
+
+        # fill_value = data.attrs['FillValue']
+        data = data.where((data >= data.attrs['valid_range'][0]) &
+                          (data <= data.attrs['valid_range'][1]))
+
+        # calibration
+        calibration = ds_info['calibration']
+
+        if calibration == 'counts':
+            return data
+        elif calibration == 'reflectance':
+            logger.debug("Calibrating to reflectances")
+            # using the corresponding SCALE and OFFSET
+            cal_coef = 'CALIBRATION_COEF(SCALE+OFFSET)'
+            slope = self.get(cal_coef)[:, 0][int(file_key[-2:])-1].values
+            offset = self.get(cal_coef)[:, 1][int(file_key[-2:])-1].values
+            data = self.dn2reflectance(data, slope, offset)
+        elif calibration == 'brightness_temperature':
+            logger.debug("Calibrating to brightness_temperature")
+            # the value of dn is the index of brightness_temperature
+            data = self.calibrate(data, 'CAL'+file_key[3:])
+
+        data.attrs.update({'platform_name': self['/attr/Satellite Name'],
+                           'sensor': self['/attr/Sensor Identification Code']})
+        data.attrs.update(ds_info)
+
+        if data.attrs.get('calibration') == 'brightness_temperature':
+            data.attrs.update({'units': 'K'})

```

Comment:
Is there a good reason this can't be put in to the if statements above that do the actual calibration? You can also do `data.attrs['units'] = 'K'`.